src/lib/services/lsp3-account.service.ts

```diff
@@ -259,31 +261,38 @@ export async function setData(
 
   const erc725Account = new UniversalProfile__factory(signer).attach(erc725AccountAddress);
 
-  let controllerAddress: string;
-  let signerPermissions: string;
+  let signersAddresses: string[];
+  let signersPermissions: string[];
 
-  if (typeof controllerAddresses[0] === 'string') {
-    controllerAddress = controllerAddresses[0];
-  } else {
-    controllerAddress = controllerAddresses[0].address;
-    signerPermissions = controllerAddresses[0].permissions;
+  if (controllerAddresses.every((address) => address as ControllerOptions)) {
```

Comment:
Do we need this check? You are checking that for every address, the address is defined. Does this not always equate to true? If the array is empty and there are no addresses then there is nothing to check so this would still be true. I think you can remove this and go straight to the map

Reply:
The address inside controllerAddresses could also be a string not ControllerOptions. We should handle both cases as we don't know what will be passed in